src/Cookie.php

```diff
@@ -37,91 +58,103 @@ final class Cookie
+    private ?bool $httpOnly = null;
 
     /**
-     * @var string SameSite prevents the browser from sending this cookie along with cross-site requests.
-     * Please note that this feature is only supported since PHP 7.3.0
-     * For better security, an exception will be thrown if `sameSite` is set while using an unsupported version of PHP.
-     * To use this feature across different PHP versions check the version first. E.g.
-     * ```php
-     * $cookie->sameSite = PHP_VERSION_ID >= 70300 ? yii\web\Cookie::SAME_SITE_LAX : null,
-     * ```
-     * @see https://www.owasp.org/index.php/SameSite for more information about sameSite.
+     * @var string|null SameSite prevents the browser from sending this cookie along with cross-site requests.
+     * @see https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#samesite-cookie-attribute for more information about sameSite.
      */
-    private string $sameSite = self::SAME_SITE_LAX;
+    private ?string $sameSite = null;
 
-    public function __construct(string $name, string $value)
+    public function __construct(string $name, string $value = '', bool $safeDefaults = true)
     {
-        // @see https://tools.ietf.org/html/rfc6265#section-4
-        // @see https://tools.ietf.org/html/rfc2616#section-2.2
-        if (!preg_match('~^[a-z0-9._\-]+$~i', $name)) {
-            throw new \InvalidArgumentException("The cookie name \"$name\" contains invalid characters.");
+        if (!preg_match(self::TOKEN, $name)) {
+            throw new InvalidArgumentException("The cookie name \"$name\" contains invalid characters.");
         }
 
         $this->name = $name;
-        $this->value = $value;
+        $this->setValue($value);
+
+        if ($safeDefaults) {
+            $this->path = '/';
+            $this->secure = true;
+            $this->httpOnly = true;
+            $this->sameSite = self::SAME_SITE_LAX;
+        }
     }
 
-    public function domain(string $domain): self
+    public function getName(): string
     {
-        $new = clone $this;
-        $new->domain = $domain;
-        return $new;
+        return $this->name;
     }
 
-    public function validFor(\DateInterval $dateInterval): self
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    public function expireAt(DateTimeInterface $dateTime): self
     {
-        $expireDateTime = (new \DateTimeImmutable())->add($dateInterval);
         $new = clone $this;
-        $new->expire = (int)$expireDateTime->format('U');
+        $new->expire = $dateTime->format('D, d M Y H:i:s T');
         return $new;
     }
 
-    public function expireAt(\DateTimeInterface $dateTime): self
+    public function validFor(DateInterval $dateInterval): self
+    {
+        $expireDateTime = (new DateTimeImmutable())->add($dateInterval);
+        return $this->expireAt($expireDateTime);
+    }
+
+    public function expireWhenBrowserIsClosed(): self
```

Comment:
Is it really needed? We have already `null` value when make new instance of Cookie

Reply:
I see this one as a way to modify already existing cookie and removing expire header from it. This behavior can't be achieved by setting expire date to anything else. The only other way would be to create a new cookie, set all properties accordingly without setting expiry.